lib/server.js

```diff
@@ -117,15 +120,55 @@ bedrock.events.on('bedrock-express.configure.routes', app => {
     routes.notify, brRest.when.prefers.ld,
     validate('continuity-server.notification'),
     asyncHandler(async (req, res) => {
-      // FIXME: we need to authenticate `peerId`; http-signatures should be
-      // added here
+      const {verified, keyId} = await _verifyRequest({req});
+      if(!(verified && req.body.peer.id === keyId)) {
+        return res.status(403).end();
+      }
       // queue handling notification and immediately send response, do not
       // wait for notification handling to complete
       _handleNotification({req}).catch(_logNotificationError);
       res.status(204).end();
     }));
 });
 
+async function _verifyRequest({req}) {
+  try {
+    const {url, method, headers} = req;
+    const expectedHeaders = [
+      '(created)',
+      '(expires)',
+      // (key-id) will include the peerId in the signature itself
+      '(key-id)',
+      '(request-target)',
+      'digest',
+      'content-type',
+      'host'
+    ];
+    const parsed = parseRequest(
+      {url, method, headers}, {headers: expectedHeaders});
+    // get parsed parameters from from HTTP header and generate signing string
+    const {keyId, signingString, params: {signature: b64Signature}} = parsed;
+    // returns the key bytes
+    const key = new Ed25519VerificationKey2018({
```

Comment:
Note: There's a `_getPublicKey` helper in signature.js (once this is moved) that may help here.